docs/DeveloperGuide.md

```diff
@@ -1,38 +1,416 @@
+
+The sequence diagram below models the interactions between various entities within the Parser component and
+the Command component when any user input is provided to SplitLah.
+<br>
+<br>
+![Reference Frame Parser Sequence Diagram](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/RefParser.drawio.png)
+<br>
+<br>
+1. When `SplitLah` reads a user input, `SplitLah` calls the `Parser#getCommand` method and passes the
+   user input as the argument.
+2. Given the user input, `Parser` class first decomposes the user input into two separate components, the _command type_
+   and the _remaining arguments_.
+   This is done using the two methods `Parser#getCommandType` and `Parser#getRemainingArguments` respectively.<br>
+   Where the input is `session /create /n Class Outing /d 15-03-2022 /pl Alice Bob`, the _command type_ is parsed as
+   `session /create` and the _remaining arguments_ is parsed as `/n Class Outing /d 15-03-2022 /pl Alice Bob`.
+   1. If the _command type_ is of invalid syntax, the method `Parser#getCommandType` returns null.
+      When null is returned, `Parser` class creates and returns an `InvalidCommand` object to `SplitLah`.
+   2. Next, to check whether the _command type_ and _remaining arguments_ are valid, `Parser` class calls the method
+      `Parser#checkIfCommandIsValid`. If either the _command type_ or the _remaining arguments_ are invalid, an error
+      message is returned by the method, which is then used to return an `InvalidCommand` object to `SplitLah`.
+3. Depending on the _command type_, `Parser` class instantiates an appropriate `XYZCommandParser` object. For example,
+   for a _command type_ of `"session /create"`, a `SessionCreateCommandParser` object is instantiated.
+   If `Parser` class does not recognise the _command type_, an `InvalidCommand` object is created and returned immediately.
+4. With the corresponding `XYZCommandParser` object instantiated, `Parser` class calls the `getCommand` method
+   of `XYZCommandParser`. This process is explained in further detail in the sequence diagrams below.
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+   ![Reference Frame Command Parser Sequence Diagram](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/RefCommandParser.drawio.png)
 
-## Design & implementation
+   ![Reference Frame ParseABC Sequence Diagram](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/RefParseABC.drawio.png)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+   ![Reference Frame InvalidCommand Instantiation Sequence Diagram](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/RefInvalidCommand.drawio.png)
```

Comment:
Diagrams provided are not explicitly linked to text explaining it. Perhaps you can consider explaining each diagram with text right below the diagram, or name the diagrams and refer to them explicitly when explaining the respective process in text.